block/block.go

```diff
@@ -122,7 +122,15 @@ func (m *Manager) applyBlock(block *types.Block, commit *types.Commit, blockMeta
 				m.logger.Error("pruning channel full. skipping pruning", "retainHeight", retainHeight)
 			}
 		}
-
+		if responses.EndBlock.RollappParamUpdates != nil {
+			newDRS := m.DRSVersionHistory.AddDRSVersion(block.Header.Height, responses.EndBlock.RollappParamUpdates.Version)
+			if newDRS {
+				_, err = m.Store.SaveDRSVersionHistory(m.DRSVersionHistory, nil)
```

Comment:
I think we should simply save DRS per height in the store (so the key is per height, similiar to how we save the proposer changes. And prune it when necessary. it will simplify all the DRS management and validation imo)

Reply:
done

Reply:
done